setup.py

```diff
@@ -19,39 +19,19 @@
 import os
 from distutils.core import setup, Extension
 from distutils.command.build import build
+from shutil import which
 import subprocess
 import versioneer
 
 
-def find_glpk_header():
-    if os.environ.get("GLPK_HEADER_PATH", None) and os.path.isdir(os.environ.get("GLPK_HEADER_PATH", None)):
-        print('glpk.h found in GLPK_HEADER_PATH environment variable')
-        glpk_header_path = os.path.join(os.environ.get("GLPK_HEADER_PATH", None), 'glpk.h')
-    elif os.path.isfile('glpk.h'):
-        print('glpk.h found in source directory')
-        glpk_header_path = os.path.join(os.getcwd(), 'glpk.h')
-    else:
-        print('Trying to determine glpk.h location')
-        glpsol_dirname = os.path.dirname(subprocess.check_output(['which', 'glpsol']))
-        glpk_header_path = os.path.join(os.path.dirname(glpsol_dirname).decode("utf-8"), 'include', 'glpk.h')
-    if os.path.exists(glpk_header_path):
-        print('glpk.h found at {}'.format(glpk_header_path))
-        return os.path.dirname(os.path.abspath(glpk_header_path))
-    else:
-        raise Exception('Could not find glpk.h! Maybe glpk or glpsol is not installed.')
-
-
-try:
-    with open('README.rst', 'r') as f:
-        long_description = f.read()
-except Exception:
-    long_description = ''
-
-
-
-glpk_header_dirname = find_glpk_header()
-
-custom_cmd_class = versioneer.get_cmdclass()
+ENV_GLPK_HEADER_PATH = 'GLPK_HEADER_PATH'
+GLPK_HEADER_NAME = 'glpk.h'
+GLPSOL_BINARY_NAME = 'glpsol'
+INCLUDE_DIRS = (
```

Comment:
Can you make those OS agnostic by using pathlib or os.path.join?

Reply:
Not sure what the proper way would be as
1. `os.path.join` does not add the leading `/` and writing `os.path.join(os.path.sep, …)` seems just as platform-dependent,
2. even the [pathlib examples](https://docs.python.org/3/library/pathlib.html) input posix paths via verbatim `/`,
2. these are *nix-specific in any case.